WeatherApp/Modules/Weather/Interactor/WheatherInteractor.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  WheatherInteractor.swift
+//  WeatherApp
+//
+//  Created by Damir Zaripov on 18.03.2018.
+//  Copyright © 2018 iOSLab. All rights reserved.
+//
+
+import Foundation
+import GooglePlaces
+
+class WheatherInteractor: WheatherInteractorInput {
+
+    weak var presenter: WheatherInteractorOutput!
+    var wheaterApiManager: WheatherAPIManagerProtocol!
+    
+    let sight = " достопримечательность"
+ 
+    func getAllInfo(from place: GMSPlace) {
+        self.getCityName(from: place)
+        self.getCityPhoto(from: place)
+        self.getWheaterInfo(from: place)
+    }
+    
+    
+    func getCityPhoto(from place: GMSPlace) {
+        let cityName = place.name
+        let queryString = cityName + sight
+        wheaterApiManager.getFirstImage(with: queryString) { [weak self] (image) in
+            DispatchQueue.main.async { [weak self] in
+                guard let strongSelf = self else { return }
+                strongSelf.presenter.didFinishingLoadingPhoto(with: CityPhoto(photo: image))
+            }
+        }
+    }
+    
+    func getWheaterInfo(from place: GMSPlace) {
+        let coordinates = Coordinates(longitude: place.coordinate.latitude, latitude: place.coordinate.longitude)
+        wheaterApiManager.getWheaterInfo(with: coordinates) { (weatherInfo) in
+            DispatchQueue.main.async { [weak self] in
+                guard let temperatureInFahrenheit = weatherInfo.hourly?.data?.first?.temperature else { return }
+                let temperatureInСesium = ((temperatureInFahrenheit - 32) / 2) + (((temperatureInFahrenheit - 32) / 2) / 10)
```

Comment:
"- Harry, do u know what kind of magic here? "
"- Yes, this is magical constants magic, It is one of the most powerful and sinister spells known to wizardkind. "